lightning/src/ln/functional_test_utils.rs

```diff
@@ -2295,19 +2332,29 @@ pub fn do_pass_along_path<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>, expected_p
 	event
 }
 
-pub fn pass_along_path<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>, expected_path: &[&Node<'a, 'b, 'c>], recv_value: u64, our_payment_hash: PaymentHash, our_payment_secret: Option<PaymentSecret>, ev: MessageSendEvent, payment_claimable_expected: bool, expected_preimage: Option<PaymentPreimage>) -> Option<Event> {
-	do_pass_along_path(origin_node, expected_path, recv_value, our_payment_hash, our_payment_secret, ev, payment_claimable_expected, true, expected_preimage)
+pub fn pass_along_path<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>, expected_path: &[&Node<'a, 'b, 'c>], recv_value: u64, our_payment_hash: PaymentHash, our_payment_secret: Option<PaymentSecret>, ev: MessageSendEvent, payment_claimable_expected: bool, expected_preimage: Option<PaymentPreimage>, is_probe: bool) -> Option<Event> {
+	do_pass_along_path(origin_node, expected_path, recv_value, our_payment_hash, our_payment_secret, ev, payment_claimable_expected, true, expected_preimage, is_probe)
 }
 
-pub fn pass_along_route<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>, expected_route: &[&[&Node<'a, 'b, 'c>]], recv_value: u64, our_payment_hash: PaymentHash, our_payment_secret: PaymentSecret) {
+pub fn pass_along_route<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>, expected_route: &[&[&Node<'a, 'b, 'c>]], recv_value: u64, our_payment_hash: PaymentHash, our_payment_secret: PaymentSecret, is_probe: bool) {
 	let mut events = origin_node.node.get_and_clear_pending_msg_events();
 	assert_eq!(events.len(), expected_route.len());
+
+	if is_probe {
+		check_added_monitors!(origin_node, expected_route.len());
+	}
+
 	for (path_idx, expected_path) in expected_route.iter().enumerate() {
 		let ev = remove_first_msg_event_to_node(&expected_path[0].node.get_our_node_id(), &mut events);
 		// Once we've gotten through all the HTLCs, the last one should result in a
-		// PaymentClaimable (but each previous one should not!), .
-		let expect_payment = path_idx == expected_route.len() - 1;
-		pass_along_path(origin_node, expected_path, recv_value, our_payment_hash.clone(), Some(our_payment_secret), ev, expect_payment, None);
+		// PaymentClaimable (but each previous one should not!). Probes should never result in a PaymentClaimable event.
+		let expect_payment = path_idx == expected_route.len() - 1 && !is_probe;
+		pass_along_path(origin_node, expected_path, recv_value, our_payment_hash.clone(), Some(our_payment_secret), ev, expect_payment, None, is_probe);
+
+		if is_probe {
+			let nodes_to_fail_payment: Vec<_> = vec![origin_node].into_iter().chain(expected_path.iter().cloned()).collect();
+			fail_payment_along_path(nodes_to_fail_payment.as_slice());
```

Comment:
Looked into `pass_failed_payment_back`. I think it will take too many changes to that function to accommodate for our use case - monitor checks and event checking seem to have a different sequence. Happy to look into a bit deeper if you think it's a hard requirement, but I personally would opt for `fail_payment_along_path`.